frontend/src/contexts/metamask/provider.js

```diff
@@ -0,0 +1,110 @@
+import { useState } from 'react';
+import detectEthereumProvider from "@metamask/detect-provider";
+import metamaskContext from './context.ts';
+import { toast } from 'react-toastify';
+
+const MetamaskContextProvider = ({ children }) => {
+    const [metamaskWeb3, setMetamaskWeb3] = useState({
+        provider: null,
+        walletAddress: null
+    })
+
+    const connectMetamask = async(setIsConnectingMetamask) => {
+        try{
+            setIsConnectingMetamask(true);
+            const provider = await detectEthereumProvider(); // provider = window.ethereum
+           
+            if(!provider) {
+                //Metamastk is not detected i.e. extension is uninstalled or corrupted
+                setIsConnectingMetamask(false);
+                toast.error("Please install Metamask!"); 
+                return false;
+            }
+
+            const walletAddress = await provider.request({ method: 'eth_requestAccounts' });
+            setMetamaskWeb3({ provider, walletAddress: walletAddress.pop().toLowerCase() })
+            setIsConnectingMetamask(false);
+
+            return true;
+        }catch(err){
+            setIsConnectingMetamask(false);
+            toast.error(err?.message || "Something went wrong!");
+            return false;
+        }
+    }
+
+    const signInUser = async (selectedVoteOption) => {
+        try {
+
+            if(!metamaskWeb3.provider){
+                toast.error("Wallet not connected");
+                return false;
+            }
+
+            const domain = {
+                name: 'Oneiro',
+                version: '1',
+                chainId: 1,
+                verifyingContract: '0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC'
```

Comment:
There should not be a verifying contract because we don't have a smart contract to verify the cote